openedx_events/event_bus/avro/tests/test_deserializer.py

```diff
@@ -0,0 +1,143 @@
+"""Tests for avro.deserializer"""
+from datetime import datetime
+from unittest import TestCase
+
+from opaque_keys.edx.keys import CourseKey
+
+from openedx_events.event_bus.avro.deserializer import AvroSignalDeserializer
+from openedx_events.event_bus.avro.tests.test_utilities import (
+    EventData,
+    NestedAttrsWithDefaults,
+    NestedNonAttrs,
+    NonAttrs,
+    SimpleAttrsWithDefaults,
+    SpecialDeserializer,
+    SubTestData0,
+    SubTestData1,
+    create_simple_signal,
+)
+
+
+class TestAvroSignalDeserializer(TestCase):
+    """Test AvroSignalDeserializer"""
+
+    def test_convert_dict_to_event_data(self):
+        SIGNAL = create_simple_signal({"test_data": EventData})
+        initial_dict = {
+            "test_data": {
+                "course_id": "bar.course",
+                "sub_name": "foo",
+                "sub_test_0": {"course_id": "a.nother.course", "sub_name": "a.sub.name"},
+                "sub_test_1": {
+                    "course_id": "b.uber.another.course",
+                    "sub_name": "b.uber.sub.name",
+                },
+            }
+        }
+        deserializer = AvroSignalDeserializer(SIGNAL)
+
+        event_data = deserializer.from_dict(initial_dict)
+
+        test_data = event_data["test_data"]
+        self.assertIsInstance(test_data, EventData)
+        self.assertEqual(test_data.course_id, "bar.course")
+        self.assertEqual(test_data.sub_name, "foo")
+
+        sub_0 = test_data.sub_test_0
+        self.assertIsInstance(sub_0, SubTestData0)
+        self.assertEqual(sub_0.course_id, "a.nother.course")
+        self.assertEqual(sub_0.sub_name, "a.sub.name")
+
+        sub_1 = test_data.sub_test_1
+        self.assertIsInstance(sub_1, SubTestData1)
+        self.assertEqual(sub_1.course_id, "b.uber.another.course")
+        self.assertEqual(sub_1.sub_name, "b.uber.sub.name")
+
+    def test_default_datetime_extension_deserialization(self):
+        SIGNAL = create_simple_signal({"birthday": datetime})
+        deserializer = AvroSignalDeserializer(SIGNAL)
+        birthday = datetime(year=1989, month=9, day=6)
+        as_dict = {"birthday": birthday.isoformat()}
+        event_data = deserializer.from_dict(as_dict)
+        birthday_deserialized = event_data["birthday"]
+        self.assertIsInstance(birthday_deserialized, datetime)
+        self.assertEqual(birthday_deserialized.year, 1989)
```

Comment:
You can't just assert that these are equal, or that they format to the same string, or some such one liner that is more complete?